be/src/exec/pipeline/exchange/local_exchange.cpp

```diff
@@ -159,6 +172,52 @@ Status PartitionExchanger::accept(const ChunkPtr& chunk, const int32_t sink_driv
     return Status::OK();
 }
 
+Status KeyPartitionExchanger::accept(const ChunkPtr& chunk, const int32_t sink_driver_sequence) {
+    size_t num_rows = chunk->num_rows();
+    size_t source_op_cnt = _source->get_sources().size();
+
+    if (num_rows == 0) {
+        return Status::OK();
+    }
+
+    auto& partitions_columns = _channel_partitions_columns[sink_driver_sequence];
+    for (size_t i = 0; i < partitions_columns.size(); ++i) {
+        ASSIGN_OR_RETURN(partitions_columns[i], _partition_expr_ctxs[i]->evaluate(chunk.get()))
+        DCHECK(partitions_columns[i] != nullptr);
+    }
+
+    Partition2RowIndexes partition_row_indexes;
+    for (int i = 0; i < num_rows; ++i) {
+        auto partition_key = std::make_shared<PartitionKey>(std::make_shared<Columns>(partitions_columns), i);
+        if (partition_row_indexes.find(partition_key) == partition_row_indexes.end()) {
```

Comment:
keep the result of find to avoid accessing the same key twice.

Reply:
done

Reply:
done